builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+        def credsId = "jazz-${randomString}";
+        sh "aws configure set profile.${credsId}.region ${configLoader.AWS.REGION}"
+        sh "aws configure set profile.${credsId}.aws_access_key_id $AWS_ACCESS_KEY_ID"
+        sh "aws configure set profile.${credsId}.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+
+        try {
+          sh "serverless package --stage ${environment_logical_id} -v > output.log"
+        } catch(ex) {
+          def outputLog = readFile('output.log').trim()
+          echo "serverless packaging log $outputLog"
+          throw ex
+        }
+
+        def outputLog = readFile('output.log').trim()
+        echo "serverless packaging log $outputLog"
+
+        def cft = readFile('./.serverless/cloudformation-template-update-stack.json').trim()
+        echo "CFT: $cft"
+      }
+
+    }
+
+    stage('Template validation') {
+      echo 'The template validation state'
+      withCredentials([
+        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']
+      ]) {
+        def randomString = utilModule.generateRequestId();
+        def credsId = "jazz-${randomString}";
+        sh "aws configure set profile.${credsId}.region ${configLoader.AWS.REGION}"
+        sh "aws configure set profile.${credsId}.aws_access_key_id $AWS_ACCESS_KEY_ID"
+        sh "aws configure set profile.${credsId}.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+
+        try {
+          sh "aws cloudformation validate-template --template-body file://.serverless/cloudformation-template-update-stack.json --region ${configLoader.AWS.REGION} > output.log"
+        } catch(ex) {
+          def outputLog = readFile('output.log').trim()
+          echo "template validation log $outputLog"
+          throw ex
+        }
+        def outputLog = readFile('output.log').trim()
+        echo "template validation log $outputLog"
+
+        def cft = readFile('./.serverless/cloudformation-template-update-stack.json').trim()
+        def cftJson = parseJson(cft)
+        def outstandingResources = whiteListModule.validate(cftJson)
+        if(!outstandingResources.isEmpty()) throw new Exception("Invalid CloudFormation template. The resource types not allowed: ${outstandingResources}")
+      }
+    }
+
+    stage("Deployment to ${env_key} environment") {
+      events.sendStartedEvent('DEPLOY_TO_AWS', 'Deployment started to staging AWS environment', context_map, environment_logical_id)
+      events.sendStartedEvent('UPDATE_ENVIRONMENT', "Environment status update event for ${env_key} deployment", environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment_logical_id, null), environment_logical_id)
+
+      echo "starts Deployment to ${env_key} Env"
+      def lambdaARN = null
+      environmentDeploymentMetadata.setEnvironmentEndpoint(lambdaARN)
```

Comment:
Remove. We might have multiple ARNs/endpoints coming out of `sls-app` deployments. We can't publish a specific endpoint per service.

Reply:
Addressed with commit: 'Removing  environmentDeploymentMetadata.setEnvironmentEndpoint(lambdaARN) as we may have the multiple ARNs/endpoints coming out of sls-app deployments'

Reply:
Resolved with https://github.com/tmobile/jazz/pull/829/commits/3db02d4a23f87f5a0ee309064407120924d0ebca